### DISCUSSION TITLE
How to add custom attributes to traces?

### DISCUSSION THREAD

**Discussion by KevinGuancheDarias** [open-telemetry/opentelemetry-java#5710] [Q&A]
Hello I have a Spring boot app, I just added the javaagent, and the application is sending a lot of useful information, the problem is that I need to be able to add custom attribute to the data, in the current case the username.

How can I archieve that?, I already tried with Span.current() (doesn't return the trace that should has been generated by the javaagent), I also tried with SpanProcessor (I don't know how to add it to the javaavent), AutoConfigurationCustomizer is not fired neither

I was expected to do something like

```java

class SomeFilter implements Filter { 

@Override
    public void doFilter(ServletRequest request, ServletResponse response, FilterChain chain) throws IOException, ServletException {
        // Add the logged user to the trace here
        chain.doFilter(request,response);
    }

}
```

Thanks in advance!

**Comment by trask** [comment]
I would expect `Span.current().setAttribute("my-attr", "my-val")` to work inside of `doFilter()`, are there any more details you can provide that could help us understand why it's not working for you?

(`SpanProcessor` is a whole different story and more complex with the Java agent, see https://github.com/open-telemetry/opentelemetry-java-instrumentation/tree/main/examples/extension#extensions)

**Comment by KevinGuancheDarias** [comment]
Hello @trask , first of all thanks for your time, it's nor working the Span.current() not only in the Filter, nor it works in the endpoint, I don't know why, it returns an invalid trace, the APM is properly registering the call to the endpoint

![image](https://github.com/open-telemetry/opentelemetry-java/assets/6832422/25e66769-5fdc-495c-a3e0-c6644229cfef)

Lol, someone told me they had only 1 dep in the pom, and then it worked, I let it reported here, because maybe others has the same problem

I had the following deps

```xml
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api-trace</artifactId>
            <version>0.13.1</version>
        </dependency>

        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-sdk</artifactId>
            <version>1.24.0</version>
        </dependency>
        <dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>1.28.0</version>
        </dependency>

```

After removing the unrequired deps and let the pom like this

```xml

<dependency>
            <groupId>io.opentelemetry</groupId>
            <artifactId>opentelemetry-api</artifactId>
            <version>1.28.0</version>
        </dependency>

```

It now works properly! 🍷 

![image](https://github.com/open-telemetry/opentelemetry-java/assets/6832422/eed1ead6-796c-468a-a79a-f463439abe6f)
